ui/src/components/uploader/uploader-core.js

```diff
@@ -442,30 +443,24 @@ export function getRenderer (getPlugin) {
     abort: state.abort
   }
 
-  // TODO: the result of this computed, especially the dynamic part, isn't currently typed
-  // This result in an error with Volar when accessing the state (eg. files array)
-  const slotScope = computed(() => {
-    const acc = {
-      canAddFiles: canAddFiles.value,
-      canUpload: canUpload.value,
-      uploadSizeLabel: uploadSizeLabel.value,
-      uploadProgressLabel: uploadProgressLabel.value
-    }
+  // expose public methods
+  Object.assign(proxy, publicMethods)
 
-    for (const key in state) {
-      acc[ key ] = isRef(state[ key ]) === true
-        ? state[ key ].value
-        : state[ key ]
-    }
+  injectMultipleProps(proxy, {
+    canAddFiles: () => canAddFiles.value,
+    canUpload: () => canUpload.value,
+    uploadSizeLabel: () => uploadSizeLabel.value,
+    uploadProgressLabel: () => uploadProgressLabel.value,
 
-    // TODO: (Qv3) Put the QUploader instance under `ref`
-    // property for consistency and flexibility
-    // return { ref: { ...acc, ...publicMethods } }
-    return { ...acc, ...publicMethods }
+    ...Object.entries(state).reduce((acc, [ key, val ]) => {
+      acc[ key ] = () => unref(val)
+      return acc
+    }, {})
   })
 
-  // expose public methods
-  Object.assign(proxy, publicMethods)
+  // TODO: (Qv3) Put the instance under `ref` property for consistency and flexibility
+  // computed(() => ({ ref: proxy, /* new stuff can be added if needed */ }))
+  const slotScope = computed(() => proxy)
```

Comment:
Hi, putting the proxy under computed() works, but it makes the whole proxy Object reactive. Besides the perf downside, the implications are huge and we can bump into all sorts of trouble. Better to build an object containing all the stuff then "inject" it into proxy and the slotScope computed.

Reply:
@rstoenescu #14967